query/src/main/scala/filodb/query/exec/PeriodicSamplesMapper.scala

```diff
@@ -357,9 +361,13 @@ class SlidingWindowIterator(raw: RangeVectorCursor,
     */
   private def shouldAddCurToWindow(curWindowStart: Long, cur: TransientRow): Boolean = {
     // One of the following three conditions need to hold true:
+    val insideCurWindow = if (inclusive)
+      cur.timestamp >= curWindowStart
+    else
+      cur.timestamp > curWindowStart
```

Comment:
Why was this `inclusive` special case needed ?
Sorry - I have to ask this for due diligence: Did you double confirm this? It's very odd that prom decided to have this special case.
The arg needs documentation.

Reply:
The tests pass, so, technically it's the proof. But the idea is that originally PeriodicSamplesMapper acted on ranges like [1m], while subqueries should act on something like [1m:10s]. If your raw data is at 10 seconds interval, foo[1m] will never get you more than 6 samples (60s/10s=6), however, subquery might get you 7 samples when start/end aligns with the step (ie start % step =0). That you can easily verify in prometheus, just run: foo[1m] and foo[1m:10s].

Reply:
I am seeing that in [this prom query](http://demo.robustperception.io:9090/graph?g0.expr=prometheus_http_requests_total%7Bcode%3D%22200%22%2C%20handler%3D%22%2Fapi%2Fv1%2Flabels%22%2C%20instance%3D%22demo.robustperception.io%3A9090%22%2C%20job%3D%22prometheus%22%7D%5B5m%3A1m%5D&g0.tab=1&g0.stacked=0&g0.range_input=1h&g0.end_input=2021-05-06%2016%3A00%3A00&g0.moment_input=2021-05-06%2016%3A00%3A00) which has [5m:1m] we see only 5 samples although now is aligned to the minute. 

![image](https://user-images.githubusercontent.com/5897052/118300920-512e9200-b497-11eb-93dc-da4ca64d27e0.png)

We need to double check this

Reply:
I am trying same query again now and am seeing 6 samples 😮

Reply:
the screenshot behavior is unusual, it should be 6 samples. However, generally speaking, inclusive on both sides window of subquery feature is unconventional (because it allows different number of samples in different windows). The resolution for now is to follow prometheus behavior and for that we will try to +1 the window in periodicsamplesmapper.